ipaserver/plugins/server.py

```diff
@@ -753,6 +753,9 @@ def pre_callback(self, ldap, dn, *keys, **options):
         self._ensure_last_of_role(
             pkey, ignore_last_of_role=options.get('ignore_last_of_role', False)
         )
+        with self.api.Backend.ra_securitydomain as domain_api:
+            domain_api.delete_domain(pkey, 'KRA')
+            domain_api.delete_domain(pkey, 'CA')
```

Comment:
When the topology is CA-less, this fails with:
```
# ipa server-del replica.ipa.test
Removing replica.ipa.test from replication topology, please wait...
ipa: ERROR: an internal error has occurred
```
The code must be called only when CA is enabled.

Reply:
Fixed and added a CAless test to the temp commit. Nice catch, thanks.